gallery/src/PageHeader/HostedVersionsSelect.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, ChangeEventHandler, useState } from 'react';
+import { NxFormSelect } from '@sonatype/react-shared-components';
+
+import packageJson from '../../package.json';
+
+import './HostedVersionsSelect.scss';
+
+const CURRENT_VERSION = packageJson.version;
+const HOSTED_VERSIONS_JSON_URL = 'https://gallery.sonatype.dev/versions/latest/hosted-versions.json';
```

Comment:
I'm not super comfortable with the idea that this is hardcoded especially when the gallery is often run on other hosts

Reply:
> I'm not super comfortable with the idea that this is hardcoded especially when the gallery is often run on other hosts

Since this file is being fetched at runtime anyway I'm wondering if it should even be managed by the RSC build.  What if the jenkins job that deploys gallery.sonatype.dev had a step where it appended the latest version number to an s3 bucket or something, and that buck was what go fetched here?

Reply:
> > I'm not super comfortable with the idea that this is hardcoded especially when the gallery is often run on other hosts
> 
> Since this file is being fetched at runtime anyway I'm wondering if it should even be managed by the RSC build. What if the jenkins job that deploys gallery.sonatype.dev had a step where it appended the latest version number to an s3 bucket or something, and that buck was what go fetched here?

Hmm, I guess one benefit of having it hardcoded here is that we can control or make changes to the versions listed through as part of our codebase (not entirely sure if that's entirely useful, perhaps filtering the versions we want users to be able to navigate to etc).
But I see how having the Jenkins build job handle the appending having s3 to a file is nice too. I'll reach out to Joel to explore that idea.

Reply:
> Hmm, I guess one benefit of having it hardcoded here is that we can control or make changes to the versions listed through as part of our codebase (not entirely sure if that's entirely useful, perhaps filtering the versions we want users to be able to navigate to etc).

That's part of what doesn't seem right to me.  Which versions are available is a concern of the gallery.sonatype.dev deployment specifically and not of this codebase more generally.  Really, even a list of available versions in an S3 bucket  is iffy as there's nothing really guaranteeing that that file is in sync with the reality of what versions are actually being hosted.  Ideally the list of versions would somehow be generated from what's actually available on the server

Reply:
> > Hmm, I guess one benefit of having it hardcoded here is that we can control or make changes to the versions listed through as part of our codebase (not entirely sure if that's entirely useful, perhaps filtering the versions we want users to be able to navigate to etc).
> 
> That's part of what doesn't seem right to me. Which versions are available is a concern of the gallery.sonatype.dev deployment specifically and not of this codebase more generally. Really, even a list of available versions in an S3 bucket is iffy as there's nothing really guaranteeing that that file is in sync with the reality of what versions are actually being hosted. Ideally the list of versions would somehow be generated from what's actually available on the server

Hmm, I see your point, I'll reach out to Joel and see how we can get a generated a list of of hosted versions.

Reply:
> > > Hmm, I guess one benefit of having it hardcoded here is that we can control or make changes to the versions listed through as part of our codebase (not entirely sure if that's entirely useful, perhaps filtering the versions we want users to be able to navigate to etc).
> > 
> > 
> > That's part of what doesn't seem right to me. Which versions are available is a concern of the gallery.sonatype.dev deployment specifically and not of this codebase more generally. Really, even a list of available versions in an S3 bucket is iffy as there's nothing really guaranteeing that that file is in sync with the reality of what versions are actually being hosted. Ideally the list of versions would somehow be generated from what's actually available on the server
> 
> Hmm, I see your point, I'll reach out to Joel and see how we can get a generated a list of of hosted versions.

We updated our approach! Instead of having the truth of the list of versions in the codebase, we have generate a json file from the directory listing after the last version is uploaded:

https://gallery.sonatype.dev/hosted-versions.json